tests/browser_type_test.go

```diff
@@ -18,3 +22,46 @@ func TestBrowserTypeConnect(t *testing.T) {
 	b := bt.Connect(tb.wsURL, nil)
 	b.NewPage(nil)
 }
+
+func TestBrowserTypeLaunchToConnect(t *testing.T) {
+	wsURLEnvVar := "K6_BROWSER_WS_URL"
+
+	setUp := func(wsURL string) {
+		os.Setenv(wsURLEnvVar, wsURL) //nolint:errcheck,gosec
+	}
+	tearDown := func() {
+		os.Unsetenv(wsURLEnvVar) //nolint:errcheck
+	}
+
+	vu := k6test.NewVU(t)
+	tb := newTestBrowser(t)
+	bp := newTestBrowserProxy(t, tb)
```

Comment:
I understand that the `browserProxy` helps assert that a connection has been made instead of launching a new browser, but i wonder if we could've launched a `newTestBrowser`; retrieved the ws url; set that as the env var; call `chromium.launch();`; and assert that there are no pids registered?

It could still be worth ensuring that a pid is not registered (if that is possible).

Reply:
We can do this by putting a field on the `JSModule` and doing `require.Len(t, jsMod.PidRegistry.Pids(), 0)`?

Reply:
Got it. Nevermind, I think we can change this now also.
See [ec5399b](https://github.com/grafana/xk6-browser/pull/819/commits/ec5399bfd420498d6e731f9abd77a4fe60c8b3b9) and [953dee6](https://github.com/grafana/xk6-browser/pull/819/commits/953dee6ee05e6377217d5bfb41ea08441b49bf7b).

Reply:
Yes, that would be a much better solution 👍

Reply:
Changed in [97c05d1](https://github.com/grafana/xk6-browser/pull/819/commits/97c05d19646c3247835f13cf997cbe6f4b4857d9) and [937ae00](https://github.com/grafana/xk6-browser/pull/819/commits/937ae00303651ef22d56dc3f4f7ab1d6404065ef).

Reply:
Yeah, this is a nice compliment 🙂 

I'm still not 100% sold on the idea of needing a proxy websocket server for this change. Why can't we work with something like:

```go
func TestBrowserTypeLaunchToConnect(t *testing.T) {
	vu := k6test.NewVU(t)
	tb := newTestBrowser(t)

	// Export WS URL env var
	// pointing to test browser proxy
	t.Setenv("K6_BROWSER_WS_URL", tb.wsURL)
```

The downside is that we can't assert using `require.True(t, bp.connected)`, but we're still able to assert that no new pid were registered, which is enough to suggest that `connect` was used, right?

Reply:
I guess we can, but for me the biggest verification that `launch` transitioned into `connect` is to verify that the "launched browser" was in fact not launched, but that it did `connect` to the WS URL that was set. Initially I tried doing this just using the `testBrowser` and verifying that the "new browser" WS URL was matching the one given. But this was "a mess" to do having Goja abstractions in between (which is necessary because the transition is mainly in the mapping layer). That's when I came with the idea to verify this connection from the other side, hence the proxy. Also I was hoping it could be useful for other test cases, but it's true that right now it's only serving this case.

I feel the pid registry len is a weeker/detail point to verify this functionality, but I'm happy to remove it if we think this is good enough :+1:

Reply:
Yeah, i do agree that the pid registry is a weaker method of asserting that no browser was launched. The combo of the unit test and a simpler integration test would've been good enough IMO. Happy to leave the proxy in for now, and if any other ideas crop up we can explore those at a later date.

Reply:
Do we agree on merging this then as it is?